src/game/server/gamecontext.cpp

```diff
@@ -2033,7 +2032,7 @@ void CGameContext::OnSayNetMessage(const CNetMsg_Cl_Say *pMsg, int ClientID, con
 	if(Length == 0 || (pMsg->m_pMessage[0] != '/' && (g_Config.m_SvSpamprotection && pPlayer->m_LastChat && pPlayer->m_LastChat + Server()->TickSpeed() * ((31 + Length) / 32) > Server()->Tick())))
 		return;
 
-	int GameTeam = ((CGameControllerDDRace *)m_pController)->m_Teams.m_Core.Team(pPlayer->GetCID());
+	int GameTeam = (m_pController)->m_Teams.m_Core.Team(pPlayer->GetCID());
```

Comment:
I'd suggest to use `CGameContext::GetDDRaceTeam()` API than spread this dependency on GameController, CTeams, CTeamsCore and it's API in those many places.

```suggestion
	int GameTeam = GetDDRaceTeam(pPlayer->GetCID());
```